[Outlook] (manifest) event-based activation with the JSON manifest (#3682)
OfficeDev/office-js-docs-pr: docs/outlook/autolaunch.md

Outlook on Windows uses a JavaScript file, while Outlook on the web and on the new Mac UI use an HTML file that can reference the same JavaScript file. You must provide references to both these files in the `Resources` node of the manifest as the Outlook platform ultimately determines whether to use HTML or JavaScript based on the Outlook client. As such, to configure event handling, provide the location of the HTML in the **\<Runtime\>** element, then in its `Override` child element provide the location of the JavaScript file inlined or referenced by the HTML.

# [Teams Manifest (developer preview)](#tab/jsonmanifest)

> 1. Open the manifest.json file.

Reviewer: ```suggestion
1. Open the **manifest.json** file.
```

1. Add the following object to the "extensions.runtimes" array. Note the following about this markup:

>    - The "minVersion" of the Mailbox requirement set has been set to "1.10" because the table at the top of this article specifies that this is the lowest version of the requirement set that supports the **OnNewMessageCompose** and **OnNewAppointmentCompose** events.

Reviewer: ```suggestion
   - The "minVersion" of the Mailbox requirement set is set to "1.10" because the table earlier in this article specifies that this is the lowest version of the requirement set that supports the `OnNewMessageCompose` and `OnNewAppointmentCompose` events.
```
Proposing use of backticks to match formatting in events table.

>    - The "id" of the runtime has been set to the descriptive name "autorun_runtime".

Reviewer: ```suggestion
   - The "id" of the runtime is set to the descriptive name "autorun_runtime".
```

>    - The "code" property has a child "page" property that is set to an HTML file and a child "script" property that has been set to a JavaScript file. You will create or edit these files in later steps. Office uses one of these values or the other depending on the platform.

Reviewer: ```suggestion
   - The "code" property has a child "page" property that is set to an HTML file and a child "script" property that is set to a JavaScript file. You'll create or edit these files in later steps. Office uses one of these values depending on the platform.
```

>        - Office on Windows executes the event handlers in a JavaScript-only runtime, which must load a JavaScript file directly.

Reviewer: ```suggestion
       - Office on Windows executes the event handlers in a JavaScript-only runtime, which loads a JavaScript file directly.
```

>        - Office on Mac and the web execute the handlers in a browser runtime, which must load an HTML file. That file, in turn, contains a `<script>` tag that will load the JavaScript file.

Reviewer: ```suggestion
       - Office on Mac and the web execute the handlers in a browser runtime, which loads an HTML file. That file, in turn, contains a `<script>` tag that loads the JavaScript file.
```

     For more information, see [Runtimes in Office Add-ins](../testing/runtimes.md).

>    - The "lifetime" property is set to "short" which means that the runtime starts up when one of the events is triggered and shuts down when the handler completes. (In certain rare cases, the runtime shuts down before the handler completes. See [Runtimes in Office Add-ins](../testing/runtimes.md).)

Reviewer: ```suggestion
   - The "lifetime" property is set to "short", which means that the runtime starts up when one of the events is triggered and shuts down when the handler completes. (In certain rare cases, the runtime shuts down before the handler completes. See [Runtimes in Office Add-ins](../testing/runtimes.md).)
```

>    - There are two types of "actions" that can run in the runtime. You will create functions to correspond to these actions in a later step.

Reviewer: ```suggestion
   - There are two types of "actions" that can run in the runtime. You'll create functions to correspond to these actions in a later step.
```

    ```json
     {
        "requirements": {
            "capabilities": [
                {
                    "name": "Mailbox",
                    "minVersion": "1.10"
                }
            ]
        },
        "id": "autorun_runtime",
        "type": "general",
        "code": {
            "page": "https://localhost:3000/commands.html",
            "script": "https://localhost:3000/launchevent.js"
        },
        "lifetime": "short",
        "actions": [
            {
                "id": "onNewMessageComposeHandler",
                "type": "executeFunction",
                "displayName": "onNewMessageComposeHandler"
            },
            {
                "id": "onNewAppointmentComposeHandler",
                "type": "executeFunction",
                "displayName": "onNewAppointmentComposeHandler"
            }
        ]
    }
    ```

1. Add the following "autoRunEvents" array as a property of the object in the "extensions" array.

    ```json
    "autoRunEvents": [
    
    ]
    ```

> 1. Add the following object to the "autoRunEvents" array. The "events" property maps handlers to events as described in the table near the top of this article. The handler names must match those used in the "actions" array in an earlier step.

Reviewer 2: Can we use bold formatting here instead of quotation marks for emphasis?

Author: I'm afraid not. We have to be consistent with how Teams and other MetaOS-related doc teams refer to the parts of the JSON manifest.

> 1. Add the following object to the "autoRunEvents" array. The "events" property maps handlers to events as described in the table near the top of this article. The handler names must match those used in the "actions" array in an earlier step.

Reviewer: ```suggestion
1. Add the following object to the "autoRunEvents" array. The "events" property maps handlers to events as described in the table earlier in this article. The handler names must match those used in the "actions" array in an earlier step.
```
